presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1634,4 +1643,31 @@ protocol::NodeStatus PrestoServer::fetchNodeStatus() {
   return nodeStatus;
 }
 
+void PrestoServer::registerDynamicFunctions() {
+  std::error_code
+      ec; // For using the non-throwing overloads of functions below.
+  auto systemConfig = SystemConfig::instance();
+  fs::path pluginDir = std::filesystem::current_path().append("plugin");
+  if (!systemConfig->pluginDir().empty()) {
+    pluginDir = systemConfig->pluginDir();
+  }
+  if (!fs::is_directory(pluginDir, ec)) {
+    PRESTO_STARTUP_LOG(INFO)
+        << "Plugin directory path: " << pluginDir
+        << " is invalid. Dynamic libraries will not be loaded on this run.";
+    return;
+  }
+  PRESTO_STARTUP_LOG(INFO) << "Loading dynamic libraries from: " << pluginDir;
```

Comment:
Put this in an `else` block and remove the `if (fs::is_directory(pluginDir, ec)` below.